### ISSUE TITLE
Builder support for Map<String, String> as an option

### ISSUE THREAD

**Issue by Tomas-Kraus** [helidon-io/helidon#8217]
It would be nice to have support for `Config` options like
```
  properties:
    eclipselink.target-database: MySQL
    eclipselink.target-server: None
    eclipselink.logging.level: FINEST
    eclipselink.logging.level.weaver: OFF
    eclipselink.logging.parameters: true
    eclipselink.weaving: false

```
to be provided using
```
    @Option.Configured
    Map<String, String> properties();
```
in the ConfigBlueprint.

Will require to modify TypeHandlerMap to process and return whole nodes subtree (including child nodes).

**Comment by tomas-langer** [comment]
There is a method `asMap()` on `Config`
This should be generated as `config.get("key").detach().asMap().orElseGet(Map::of)`
